pkg/resourceinterpreter/default/native/retain.go

```diff
@@ -40,29 +41,43 @@ func retainPodFields(desired, observed *unstructured.Unstructured) (*unstructure
 		return nil, fmt.Errorf("failed to convert clusterPod from unstructured object: %v", err)
 	}
 
-	desiredPod.Spec.NodeName = clusterPod.Spec.NodeName
-	desiredPod.Spec.ServiceAccountName = clusterPod.Spec.ServiceAccountName
-	desiredPod.Spec.Volumes = clusterPod.Spec.Volumes
-	// retain volumeMounts in each container
-	for _, clusterContainer := range clusterPod.Spec.Containers {
-		for desiredIndex, desiredContainer := range desiredPod.Spec.Containers {
+	// Pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`,
+	// `spec.activeDeadlineSeconds`, `spec.tolerations` (only additions to existing tolerations) or
+	// `spec.terminationGracePeriodSeconds` (allow it to be set to 1 if it was previously negative).
+	// More info: https://kubernetes.io/docs/concepts/workloads/pods/#pod-update-and-replacement
+	for clusterIndex, clusterContainer := range clusterPod.Spec.InitContainers {
+		for _, desiredContainer := range desiredPod.Spec.InitContainers {
 			if desiredContainer.Name == clusterContainer.Name {
-				desiredPod.Spec.Containers[desiredIndex].VolumeMounts = clusterContainer.VolumeMounts
+				clusterPod.Spec.InitContainers[clusterIndex].Image = desiredContainer.Image
```

Comment:
I have two questions.
1. Why delete the `VolumeMounts` field?   Refer to https://github.com/karmada-io/karmada/issues/3424
2.  Can this image field be modified in the control panel?

Reply:
> Why delete the VolumeMounts field? Refer to https://github.com/karmada-io/karmada/issues/3424

No, I did not give up retaining `VolumeMounts `. You can see that `clusterPod` is returned instead of `desiredPod`.

> Can this image field be modified in the control panel?

Sure.

Reply:
> No, I did not give up retaining VolumeMounts . You can see that clusterPod is returned instead of desiredPod.

Got it.

> Can this image field be modified in the control panel?

Sorry, I didn't explain my idea clearly. 

What I meant was whether we can modify fields other than the `image` field? 

For example, the `label` field?

Reply:
Of course, you can. But you should know that _Pod_ update operations have some limitations.
More info: https://kubernetes.io/docs/concepts/workloads/pods/#pod-update-and-replacement

Reply:
> clusterPod is returned instead of `desiredPod`.

 I think the newly added `label` should be in `desiredPod`. But you returned `clusterPod`, right?

Reply:
> I think the newly added label should be in desiredPod. But you returned clusterPod, right?

It has nothing to do with this PR. You'll figure it out with function [retainClusterFields](https://github.com/karmada-io/karmada/blob/master/pkg/util/objectwatcher/objectwatcher.go#L99).

Reply:
Have you tested it?

Reply:
For example, add a new  label field.

Reply:
If add add a new label, I don't think it will propagate to the member cluster.

Reply:
Sorry, I did not get you. I know what you mean now. It does need to be fixed. Thanks.

Reply:
I have another question.

Currently, if users use an OverridePolicy to modify these fields(other than `spec.containers[*].image, spec.initContainers[*].image, spec.activeDeadlineSeconds or spec.tolerations`.), users will get some error messages. but If we retain these fields, use can't get any error message. is this expected ?

Reply:
It makes no sense, I think. For example, I use an OverridePolicy to modify resource's `ownerReferences`, I also won't see any error message. People should know thoes restrictions of K8s and Karmada.